fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/impl/FHIRPersistenceJDBCImpl.java

```diff
@@ -681,7 +682,43 @@ private ParameterDAO makeParameterDAO(Connection connection) throws FHIRPersiste
                         }
                     } else {
                         unsortedResultsList = resourceDao.search(query);
-                        resources = this.convertResourceDTOList(unsortedResultsList, resourceType, elements);
+                        List<com.ibm.fhir.persistence.jdbc.dto.Resource> matchResultList = unsortedResultsList;
+                        List<com.ibm.fhir.persistence.jdbc.dto.Resource> includeResultList = new ArrayList<>();
+
+                        // Check if _include or _revinclude search. If so, remove duplicates from 'include' resources
+                        // (duplicates of both 'match' and 'include' resources) and make sure _elements processing
+                        // is not done for 'include' resources.
+                        if (searchContext.hasIncludeParameters() || searchContext.hasRevIncludeParameters()) {
+                            // Calculate 'match' results count
+                            int pageSize = searchContext.getPageSize();
+                            int offset = (searchContext.getPageNumber() - 1) * pageSize;
+                            int matchResultCount = pageSize;
+                            if (searchResultCount < offset + pageSize) {
+                                matchResultCount = searchResultCount - offset;
+                            }
+
+                            // Split results list into 'match' list and 'include' list
+                            if (unsortedResultsList.size() > matchResultCount) {
+                                matchResultList = unsortedResultsList.subList(0, matchResultCount);
```

Comment:
I'm confused by the variable name `unsortedResultsList`.  If its unsorted, then how do we know the `match` resources come first?

Reply:
This is a holdover from the existing code. It's unsorted in that the _sort parameter was not specified for this search. However, the match results are ensured to be sorted before the include results based on the SQL sort I implemented for issue #1494.

Reply:
I renamed this variable to avoid confusion, and added a comment in the section of code where I'm splitting the results to clarify that the 'match' results have been sorted ahead of the 'include' results.